datadog/tests/resource_datadog_rum_retention_filters_order_test.go

```diff
@@ -0,0 +1,101 @@
+package test
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-testing/helper/resource"
+	"github.com/hashicorp/terraform-plugin-testing/terraform"
+
+	"github.com/terraform-providers/terraform-provider-datadog/datadog/fwprovider"
+)
+
+const RumRetentionFiltersOrderResourceTestAppId = "17a2877d-5a77-406e-9039-9da24714936e"
+
+func validDatadogRumRetentionFiltersOrder() string {
+	return fmt.Sprintf(`data "datadog_rum_retention_filters" "my_retention_filters" {
+		application_id = %q
+	}
+
+	resource "datadog_rum_retention_filters_order" "testing_rum_retention_filters_order" {
+		application_id = %q
+	    retention_filter_ids = concat([
+		    data.datadog_rum_retention_filters.my_retention_filters.retention_filters[0].id
+	    ],[
+		    for rf in slice(data.datadog_rum_retention_filters.my_retention_filters.retention_filters, 1, length(data.datadog_rum_retention_filters.my_retention_filters.retention_filters)) : rf.id
+	    ])
+	}
+	`, RumRetentionFiltersOrderResourceTestAppId, RumRetentionFiltersOrderResourceTestAppId)
+}
+
+func unmatchedErrorDatadogRumRetentionFiltersOrder() string {
+	return fmt.Sprintf(`resource "datadog_rum_retention_filters_order" "testing_rum_retention_filters_order_" {
+		application_id = %q
+	    retention_filter_ids = ["not_found_uuid"]
+	}
+	`, RumRetentionFiltersOrderResourceTestAppId)
+}
+
+func TestAccRumRetentionFilterOrder(t *testing.T) {
+	t.Parallel()
+	_, providers, accProviders := testAccFrameworkMuxProviders(context.Background(), t)
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV5ProviderFactories: accProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: validDatadogRumRetentionFiltersOrder(),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDatadogRumRetentionFiltersOrderResourceMatch(providers.frameworkProvider,
```

Comment:
This test will fail if any of the default retention filters is deleted and we don't have at least 4 ids right? This sounds like it is going to be hard to maintain, maybe we should instead create a few RFs instead, put them first in a given order, and only assess the ordering on these.